mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -460,16 +460,16 @@ object LinearRegression extends DefaultParamsReadable[LinearRegression] {
   val MAX_FEATURES_FOR_NORMAL_SOLVER: Int = WeightedLeastSquares.MAX_NUM_FEATURES
 
   /** String name for "auto". */
-  private[regression] val AUTO = "auto"
+  private[regression] val Auto = "auto"
```

Comment:
@facaiy We had some offline discussion, and decided to fix some small issues in this PR.